ecl/hql/hqlgram.y

```diff
@@ -10560,6 +10566,46 @@ optSortList
     | ',' sortList
     ;
 
+optTraceFlags
+    : traceFlags
+    |                   {   $$.setNullExpr(); }
+    ;
+
+traceFlags
+    : ',' traceFlag     {   $$.setExpr($2.getExpr()); }
+    | traceFlags ',' traceFlag
+                        {   $$.setExpr(createComma($1.getExpr(), $3.getExpr())); }
+    ;
+
+traceFlag
+    : KEEP expression   {
```

Comment:
Coped from dedupFlag...

I can add the parens (i.e. make them compulsory) if you want. As it stands it works either way.